endoreg_db/services/video_import.py

```
@@ -372,294 +851,122 @@ def _ensure_frame_cleaning_available(self):
-        path = Path(self.current_video.raw_file.path)
+        raw_video_path = self.processing_context.get('raw_video_path')
+        
+        if not raw_video_path or not Path(raw_video_path).exists():
+            raise RuntimeError(f"Raw video path not found: {raw_video_path}")
         
         # Get processor name safely
-        processor = getattr(self.current_video.video_meta, "processor", None) if self.current_video.video_meta else None
+        video = self._require_current_video()
+        video_meta = getattr(video, "video_meta", None)
+        processor = getattr(video_meta, "processor", None) if video_meta else None
         device_name = processor.name if processor else self.processing_context['processor_name']
```

Ruff (0.14.2)

Local variable `device_name` is assigned to but never used

Remove assignment to unused variable `device_name`

(F841)